lib/lambda.js

```diff
@@ -199,7 +221,10 @@ function setSqsData (agent, trans, event, context, faasId, isColdStart) {
   }
 
   if (agent._conf.captureHeaders) {
-    messageContext.headers = Object.assign({}, record.messageAttributes)
+    const headers = headersFromSqsMessageAttributes(record.messageAttributes)
+    if (Object.keys(headers).length > 0) {
```

Comment:
:+1: for `Object.keys(obj)` -- I'm pretty sure that's the fastest way to do this without needing to get into `for in` and `hasOwnProperty` territory. Unrelated to this PR/note-to-self -- a custom eslint linting rule that ensures we're using the more efficient Object.keys() for object looping/iteration (vs. one of the other myriad ways you can loop an object in javascript) would be a nice to have.